server/src/mysql/worker.rs

```diff
@@ -0,0 +1,145 @@
+// Copyright 2022 CeresDB Project Authors. Licensed under Apache-2.0.
+
+use std::{marker::PhantomData, sync::Arc};
+
+use catalog::manager::Manager as CatalogManager;
+use log::{error, info};
+use opensrv_mysql::{AsyncMysqlShim, ErrorKind, QueryResultWriter, StatementMetaWriter};
+use query_engine::executor::Executor as QueryExecutor;
+use snafu::ResultExt;
+use table_engine::engine::EngineRuntimes;
+
+use crate::{
+    context::RequestContext,
+    handlers::{
+        self,
+        sql::{Request, Response},
+    },
+    instance::Instance,
+    mysql::{
+        error::{CreateContext, HandleSql, Result},
+        writer::MysqlQueryResultWriter,
+    },
+};
+
+pub struct MysqlWorker<W: std::io::Write + Send + Sync, C, Q> {
+    generic_hold: PhantomData<W>,
+    instance: Arc<Instance<C, Q>>,
+    runtimes: Arc<EngineRuntimes>,
+}
+
+impl<W, C, Q> MysqlWorker<W, C, Q>
+where
+    W: std::io::Write + Send + Sync,
+    C: CatalogManager + 'static,
+    Q: QueryExecutor + 'static,
+{
+    pub fn new(instance: Arc<Instance<C, Q>>, runtimes: Arc<EngineRuntimes>) -> Self {
+        Self {
+            generic_hold: PhantomData::default(),
+            instance,
+            runtimes,
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl<W, C, Q> AsyncMysqlShim<W> for MysqlWorker<W, C, Q>
+where
+    W: std::io::Write + Send + Sync,
+    C: CatalogManager + 'static,
+    Q: QueryExecutor + 'static,
+{
+    type Error = crate::mysql::error::Error;
+
+    async fn on_prepare<'a>(
+        &'a mut self,
+        _query: &'a str,
+        info: StatementMetaWriter<'a, W>,
+    ) -> Result<()> {
+        info.error(
+            ErrorKind::ER_UNKNOWN_ERROR,
```

Comment:
Just came across this when review recent PR. 

Isn't [ER_NOT_SUPPORTED_YET](https://docs.rs/opensrv-mysql/0.1.0/opensrv_mysql/enum.ErrorKind.html#variant.ER_NOT_SUPPORTED_YET) more suitable here?